tecton_client/_internal/data_types.py

```diff
@@ -45,16 +47,40 @@ def get_features_dict(self) -> Dict[str, Any]:
             for metadata, raw_feature_value in zip(features_metadata, self.result.features)
         }
 
-    def _get_feature_value(self, data_type, raw_feature_value):
+    def _get_feature_value(self, data_type_info, raw_feature_value):
         """Convert response value from str to int if the data type is int64"""
+        data_type = data_type_info.get("type")
         if raw_feature_value is None:
-            return raw_feature_value
-        elif data_type.get("type") == "int64":
-            return int(raw_feature_value)
-        elif data_type.get("type") == "array" and data_type.get("elementType") == "int64":
-            return map(int, raw_feature_value)
+            value = None
+        elif data_type == "int64":
+            value = self._convert_int64(raw_feature_value)
+        elif data_type == "array":
+            element_type = data_type_info.get("elementType")
+            value = [self._get_feature_value(element_type, value) for value in raw_feature_value]
+        elif data_type == "map":
+            # apply the unpacking function to the keys and values separately
+            key_type = data_type_info.get("keyType")
+            value_type = data_type_info.get("valueType")
+            value = {
+                self._get_feature_value(key_type, key): self._get_feature_value(value_type, value)
+                for key, value in raw_feature_value.items()
+            }
+        elif data_type == "struct":
+            field_meta_list = data_type_info.get("fields")
+            value = {
+                field_meta.get("name"): self._get_feature_value(field_meta.get("dataType"), field_value)
+                for field_meta, field_value in zip(field_meta_list, raw_feature_value)
+            }
+        else:
+            value = raw_feature_value
+        return value
```

Comment:
nit: a style I like better for this type of construct is returning early in the conditional (makes it clear that the result isn't b eing used later)

```
if value is None:
    return None
if data_type == "int64":
    return _convert_int64(raw_feature_value)
if data_type == "array":
    element_type = data_type_info.get("elementType")
    return [_get_feature_value(element_type, value) for value in raw_feature_value]
... etc
```